cre.cpp

```diff
@@ -3495,6 +3495,38 @@ static int getImageDataFromPosition(lua_State *L) {
     return 0;
 }
 
+static int setUserHyphenationDict(lua_State *L) {
+    CreDocument *doc = (CreDocument*) luaL_checkudata(L, 1, "credocument");
+    const char *filename = luaL_checkstring(L, 2);
+    int no_sloppy_load = luaL_checkint(L, 3);
+    UserHyphenDict::init(filename, no_sloppy_load !=0);
+    return 0;
+}
+
+static int getHyphenationForWord(lua_State *L) {
+    CreDocument *doc = (CreDocument*) luaL_checkudata(L, 1, "credocument");
+    const char *word = luaL_checkstring(L, 2);
+    lString32 hyphenation = UserHyphenDict::getHyphenation(word);
+    lua_pushstring(L, UnicodeToLocal(hyphenation).c_str());
+    return 1;
+}
+
+static int getLower(lua_State *L) {
+    CreDocument *doc = (CreDocument*) luaL_checkudata(L, 1, "credocument");
+    const char *word = luaL_checkstring(L, 2);
+    lString32 word_lower = UserHyphenDict::getLower(word);
```

Comment:
That's what I aimed at in a previous comment:
> Is formatHyphenation() to just not have to add utf8proc ffi stuff ? (which is fine as long as the function name hides this fact :)

It feels (here and in hyphman.h) strange to export such a low level lowercasing API function.
I'm OK with not wasting time with making a ffi/utf8proc.lua - and it could make sense to be sure we get the exact same lowercasing result as crengine.
So, please juste give it a less low-level name: `getLowercasedWord()` here and `UserHyphenDict::getLowercasedWord(word)` - which makes it feel it's all hyphenation higher level business.